### ISSUE TITLE
Dont re-use get_a_link transfer_option for (forced) guest_transfer_options

### ISSUE THREAD

**Issue by peter-** [filesender/filesender#591]
For "normal" use of the system by authenticated subjects it makes sense to have the `get_a_link` setting within the `transfer_options`: The person de-/selecting that checkbox is a trusted user of the system and knows best how to communicate the URL resulting from an upload to the intended recipients of the uploaded files: Either by having Filesender notify the recipients with a generic email or by the uploading subject using whatever means they prefer.  
But that setting is currently re-used for the notification of third parties -- recpients of uploads performed by guests using a voucher issued to them -- where this isn't (or isn't always) appropriate, especially since those "Created transfers options" (which I think really means "transfer options forced upon guest voucher users") cannot be changed by the voucher user, contrary to the `get_a_link` setting within the `transfer_options` where the setting originates.

E.g. within the `transfer_options` I have `get_a_link` set up this way:
```php
'get_a_link' => array(
  'available' => true,
  'advanced' => false,
  'default' => true,
),
```
so an authenticated subject can decide how they themselfs prefer to notify the intended recpients.

But that same setting and default value does not make sense when applied in the context of creating guest vouchers with forced settings: Nothing is gained by preventing the voucher user (the uploader) from using Filesender to notify any intended recpients of uploaded files via email. The uploader (voucher user) can even post the Filesender download URL to Twitter (for all I know) so why disallow them to use Filesender to do that notification?

Even worse: Other sensible settings within `guest_options` such as `can_only_send_to_me` (to make sure at least one leg of a Filesender transfer is always tied to the community running the Filesender instance) now may have lost their chosen `'default' => true` setting since the `get_a_link` setting from the `transfer_options` takes prececends over this setting (and both cannot sensibly be true at the same time) and `can_only_send_to_me` cannot be selected as long as `get_a_link` is selected (which it is by default in our case, though not intended to limit how guest voucher users notify the intended recipients of their own uploads.)

So now people creating vouchers will have to remember to
1. deselect the `Get a link instead of sending to recipients` checkbox and then
2. select the `Can only send to me` checkbox

and I cannot change this with configuration alone.

In short: It does not make sense to re-use that same setting and value also for voucher users,  particularly as they cannot change that setting themselfs.  
So I'd suggest to stop re-using that `transfer_option` within `guest_transfer_options` and either picking a sensible default (e.g. `get_a_link` `'default' => false` for guest transfers) or implement a separate setting for `guest_transfer_options`. (Or maybe don't enforce the value so that the voucher user has the same choice as authenticated subjects when uploading.)

**Comment by monkeyiq** [comment]
I think that the wording of the "Created transfers options" might need some updating along the lines you suggest. As the intent here is to bind the invited guest to the options you are setting. If you select "can only send to me" then they can only do that. On the other hand, allowing "get a link" gives the guest fairly complete access. From a security perspective if you do not select "can only send to me" then the user could enter a junk email address, or their own guest email address to send the file to in order to check the email and get a link to use anywhere anyway.

I have merged https://github.com/filesender/filesender/pull/606 which hints that "can only send to me" and "get a link" are somewhat bound together. Perhaps we should remove those entries and use a drop down selector to make this a single option which is explicit to the user. PR 606 should allow the can_only_send_to_me guest_option to play nice with get_a_link. When the can_only_send_to_me option is true from config It will initially present get_a_link unchecked. If you select get_a_link then a notification is presented letting you know that can_only_send_to_me has been unset.

**Comment by peter-** [comment]
A(nother) consequence of re-using the `get_a_link` transfer option directly as (forced) guest transfer option: I cannot have `get_a_link` set to `available` (and `default=>true`) within `transfer_options` (to encourage authenticated subjects to communicate the download URL to recipients themselfs, but still allow use of Filesender to send out those emails if desired) but at the same time **not** offer this option when creating guest vouchers (even if I have set `can_only_send_to_me` as the default there, and -- thanks to your recent changes in this area -- consequently now `get_a_link` at least defaults to `false` for guest vouchers).

Or simpler, have `get_a_link` in "Advanced settings" for guests but have `advanced=>false` for ordinary users in `transfer_options`: There's only one setting for both use-cases.

As soon as  `get_a_link` is set to `available` in `transfer_options` (where I want authenticated subjects to be able to chose) it also becomes available for guests in "Created transfers options" (where I don't want them to make that choice obvious/common since it's too complicated to understand, IMHO, and also the option's text is misleading/wrong when used as a guest option, at least in some translations: E.g. re-translating from German it states "You will recieve..." which is only true for non-guest transfer options where "You" is the authenticated Filesender user, but it's wrong for guest vouchers since it's not "me" who will receive those links but the recipient of the voucher when finishing their upload. So for guests the text would have to read "The guest/voucher recipient will recieve..." or something like that. But you can't have differing texts as... it's only one option.

The only way to avoid this confusing situation is always enforcing `can_only_send_to_me` by setting it to `default=>true` and `available=>false` within `guest_options`. Even if I would have preferred to only make it "advanced", not prohibited, for use with guest vouchers.

**Comment by peter-** [comment]
Actually, the same confusion/wording error applies to all the other transfer options when in guest mode:
* Notify *me* when/upon ...
* Send *me* ...
* etc.

In all those cases it's not "me" who's going to receive anything, it's the recipient of the guest voucher.

**Comment by peter-** [comment]
It gets worse. I recently said here:

> The only way to avoid this confusing situation is always enforcing `can_only_send_to_me` by setting it to `default=>true` and `available=>false` within `guest_options`. 

But when this is combined with `get_a_link` set `default=>true` (and `available=>true`, but that's the case only for authenticated subjects, not guests, as the get_a_link options vanishes for guests when `can_only_send_to_me` is enforced, as per above) after completion of an upload using a guest voucher NO email is generated to creator of the can-only-send-to-me-voucher (nor is a download URL shown to the guest uploader, though that part is fine).

Only if the voucher creator looks into their guests page they will see the upload under "Guests transfers".

So it seems to me the trickery done to disable get_a_link once can_only_send_to_me is set does not extend to sending notifications to the voucher creator.

Note that this is all in context of my (increasingly frustrating) attempts at actually making use of the get_a_link option so tha people will share filesender URLs themselfs instead of mis-using it as a poor man's webmail system.

**Comment by peter-** [comment]
Also I now note that running tag `filesender-2.8` with `get_a_link` and `can_only_send_to_me` both  set to true by default:
```php
$config['transfer_options'] = array(
  'get_a_link' => array(
    'available' => true,
    'advanced' => false,
    'default' => true,
  ),
);
$config['guest_options'] = array(
  'can_only_send_to_me' => array(
    'available' => false,
    'advanced' => false,
    'default' => true 
  ),
);
```
does **not** load the upload page w/ `get_a_link` selected (anymore?). Instead the email input fields (To, Subject, Message) are shown on the left and the "Get a link" checkbox (unchecked) on the right. So the UI does not match the config, IMO.

![get-a-link-true-avail](https://user-images.githubusercontent.com/4047365/65393491-7da51000-dd81-11e9-81ba-8142dba0a6aa.png)

This is on a clean clone:
```sh
$ git status
HEAD detached at filesender-2.8
nothing to commit, working tree clean
```

Only when I change `get_a_link`'s available to `false` (as the only change)
```php
$config['transfer_options'] = array(
  'get_a_link' => array(
    'available' => false,
    'advanced' => false,
    'default' => true)
);
```
is get_a_link the default when loading the upload page.
![get-a-link-true-noavail](https://user-images.githubusercontent.com/4047365/65393510-afb67200-dd81-11e9-9f21-22b234240ea5.png)
(Of course then there's no way to change that.)
